### ISSUE TITLE
`MA0138` does not account for `IAsyncEnumerable<T>`

### ISSUE THREAD

**Issue by RenderMichael** [meziantou/Meziantou.Analyzer#713]
### Version of the Meziantou.Analyzer NuGet package

2.0.149

### Rule Identifier

MA0138

### Target Framework

net8.0

### C# Language version

C# 12

### Description

`MA0138 - Do not use 'Async' suffix when a method does not return an awaitable type`

This rule still fires when a method returns `IAsyncEnumerable<T>`

### Reproduction Steps

Minimal code:

```csharp
internal async IAsyncEnumerable<string> GetValues()
{
    yield return await XYZ();
}
```

**Comment by meziantou** [comment]
I'm not sure about this one. The BCL uses both conventions. For example:

````c#
public static IAsyncEnumerable<Task<TResult>> WhenEach<TResult>(IEnumerable<Task<TResult>> tasks)
public virtual async IAsyncEnumerable<T> ReadAllAsync([EnumeratorCancellation] CancellationToken cancellationToken = default)
````

Also, the method is not directly async, but the iteration is async.

So, I would prefer a new rule for reporting `IAsyncEnumerable` methods instead of adding the behavior to the existing rule.

**Comment by RenderMichael** [comment]
I appreciate the configurability.

So this should leave the existing rule tolerant of either form for `IAsyncEnumerable<T>`?

**Comment by meziantou** [comment]
You should enable the new rule MA0157 or MA0158 to opt-in the naming convention for IAsyncEnumerable
